caikit/core/model_manager.py

```diff
@@ -251,7 +251,9 @@ def load(
         load_path = get_config().load_path
         if load_path is not None and isinstance(module_path, str):
             if not os.path.exists(module_path):
-                module_path = os.path.join(load_path, module_path)
+                full_module_path = os.path.join(load_path, module_path)
+                if os.path.exists(full_module_path):
+                    module_path = full_module_path
```

Comment:
nit: The `self._load_from_dir` after this has a misleading name now since it invokes finders + initializers that may not require that a path to a directory on disk